src/smolagents/models.py

```diff
@@ -1059,8 +1059,10 @@ def generate_stream(
         )
         for event in self.client.completion(**completion_kwargs, stream=True, stream_options={"include_usage": True}):
             if event.choices:
-                if event.choices[0].delta is None:
-                    if not getattr(event.choices[0], "finish_reason", None):
+                if event.choices[0].delta is None or event.choices[0].delta.content is None:
+                    if getattr(event.choices[0], "finish_reason", None):
+                        break  # Stop streaming if a finish reason is provided
+                    else:
                         raise ValueError(f"No content or tool calls in event: {event}")
                 else:
                     yield ChatMessageStreamDelta(
```

Comment:
And what about simplifying even further the logic?
```suggestion
                if event.choices[0].delta and event.choices[0].delta.content:
                    yield ChatMessageStreamDelta(
```

Reply:
Nice one, added that to last commit along with removing the check of `event.choices[0].delta`